src/commands/remoteDebug/startRemoteSsh.ts

```diff
@@ -0,0 +1,110 @@
+import { SiteTreeItem } from '../../explorer/SiteTreeItem';
+import { WebAppTreeItem } from '../../explorer/WebAppTreeItem';
+import { ext } from '../../extensionVariables';
+import { delay } from '../../utils/delay';
+import * as remoteDebug from './remoteDebugCommon';
+
+const remoteSsh: Map<string, boolean> = new Map();
+
+export async function startRemoteSsh(node?: SiteTreeItem): Promise<void> {
+    if (!node) {
+        node = <SiteTreeItem>await ext.tree.showTreeItemPicker(WebAppTreeItem.contextValue);
+    }
+    if (remoteSsh.get(node.root.client.fullName)) {
+        throw new Error(`Azure Remote SSH is currently starting or already started for "${node.root.client.fullName}".`);
+    }
+
+    remoteSsh.set(node.root.client.fullName, true);
+    try {
+        await startRemoteSshInternal(node);
+    } catch (error) {
+        remoteSsh.set(node.root.client.fullName, false);
+        throw error;
+    }
+}
+
+async function startRemoteSshInternal(node: SiteTreeItem): Promise<void> {
+    const siteClient: SiteClient = node.root.client;
+    const siteConfig: SiteConfigResource = await siteClient.getSiteConfig();
+    const oldSetting: boolean = <boolean>siteConfig.remoteDebuggingEnabled;
+    // should always be an unbound port
+    const portNumber: number = await portfinder.getPortPromise();
+
+    await vscode.window.withProgress({ location: vscode.ProgressLocation.Notification }, async (progress: vscode.Progress<{}>): Promise<void> => {
+        if (!siteConfig.linuxFxVersion) {
+            throw new Error('Azure Remote SSH is only supported for Linux web apps.');
+        }
+
+        remoteDebug.reportMessage('Checking app settings...', progress);
+
+        // remote debugging has to be disabled in order to tunnel to the 2222 port
+        await remoteDebug.setRemoteDebug(false, undefined /*skips confirmation*/, undefined, siteClient, siteConfig);
+
+        remoteDebug.reportMessage('Starting tunnel proxy...', progress);
+
+        const publishCredential: User = await siteClient.getWebAppPublishCredential();
+        const tunnelProxy: TunnelProxy = new TunnelProxy(portNumber, siteClient, publishCredential);
+        await callWithTelemetryAndErrorHandling('appService.remoteSshStartProxy', async function (this: IActionContext): Promise<void> {
+            this.rethrowError = true;
+            await tunnelProxy.startProxy(2222);
+            await connectToTunnelProxy(tunnelProxy);
+        });
+    });
+
+    async function connectToTunnelProxy(tunnelProxy: TunnelProxy): Promise<void> {
+        const sshTerminalName: string = `${node.root.client.fullName} - Remote SSH`;
+        // -o StrictHostKeyChecking=no doesn't prompt for adding to hosts
+        // -o "UserKnownHostsFile /dev/null" doesn't add host to known_user file
+        // -o "LogLevel ERROR" doesn't display Warning: Permanently added 'hostname,ip' (RSA) to the list of known hosts.
```

Comment:
Do we need `-o "LogLevel ERROR"` if you're not adding these to known hosts file?

Reply:
Yeah, due to the warning that says we added a RSA key to the known_hosts file (even though we actually didn't)

Reply:
Could this hide other important warnings?

Reply:
I'm not sure.  I would assume once you run the SSH command, it wouldn't affect anything else since you're technically in the other container at that point.